ecc/bls12381/g1.go

```diff
@@ -200,7 +207,10 @@ func (g *G1) Hash(msg, dst []byte) {
 
 		var x, gx, y, four ff.Fp
 		four.SetUint64(4)
-		x.SetString(xi.Text(10))
+		err := x.SetString(xi.Text(10))
+		if err != nil {
+			panic("bad string")
```

Comment:
The only reason this panic doesn't happen randomly is the hash output is shorter than the prime. Which is not obvious and deserves comment. Also this breaks  indistinguishability: x will always be smaller than 2^256, which has a low probability of happening compared to picking a point at random.

Reply:
The body of this function will be replaced entirely to follow hash2curve standard